scripts/reindex.py

```diff
@@ -96,8 +106,29 @@ def main(argv: List[str]):
 
     configure_script_logging(logger)
 
+    source_globs = set(args.sources)
+    if source_globs != {'*'} and not args.partition_prefix_length:
+        parser.error('Cannot specify sources when performing a local reindex')
+
     azul = AzulClient(num_workers=args.num_workers)
 
+    sources_by_catalog = defaultdict(set)
+    globs_matched = set()
+    for catalog in args.catalogs:
+        sources = azul.catalog_sources(catalog)
+        for source_glob in source_globs:
+            matches = fnmatch.filter(sources, source_glob)
+            if matches:
+                globs_matched.add(source_glob)
+            logger.debug('Matched sources %r in catalog %r', matches, catalog)
+            sources_by_catalog[catalog].update(matches)
+        reject(not sources_by_catalog[catalog],
```

Comment:
Doesn't quite capture my intention behind https://github.com/DataBiosphere/azul/pull/2777#discussion_r583911217

If the user specifies a very specific glob that only matches one source in one catalog, the user shouldn't have to then also specify the catalog. What's the point in doing that? The glob alone is already unambiguous. 

If a glob matches two or more sources from different catalogs (for example if two catalog use the same source) and the user doesn't want to reindex both catalogs, the user can then opt to disambiguate via `--catalogs`.